packages/core/src/components/Typography/FontPropertyRow.tsx

```diff
@@ -0,0 +1,112 @@
+import { Box, Button, Stack, Text, useDisclosure } from '@chakra-ui/react'
+import { TFontSizeVariant, TFontWeightVariant } from '@core/types'
+
+import { EditFontSizeModal } from './FontSize/EditFontSizeModal'
+import { FontSizeData } from './FontSize/FontSizeData'
+import { EditFontWeightModal } from './FontWeight/EditFontWeightModal'
+import { FontWeightData } from './FontWeight/FontWeightData'
+
+export function FontPropertyRow({
+  fontProperty,
+  fontPropertyData,
+  onUpdateFontPropertyVariant,
+  onDeleteFontPropertyVariant,
+}: {
+  fontProperty: 'fontSize' | 'fontWeight'
+  fontPropertyData: TFontSizeVariant | TFontWeightVariant
+  onUpdateFontPropertyVariant: (
+    newVariant: TFontSizeVariant | TFontWeightVariant
+  ) => void
+  onDeleteFontPropertyVariant: () => void
+}) {
+  const {
+    isOpen: isEditVariantModalOpen,
+    onOpen: onEditVariantModalOpen,
+    onClose: onEditVariantModalClose,
+  } = useDisclosure()
+
+  return (
+    <Box>
+      <Stack
+        css={{ alignItems: 'center', width: '100%' }}
+        spacing={8}
+        direction="row"
+      >
+        <Box
+          css={{
+            padding: '12px',
+            border: '1px dashed black',
+            borderRadius: 8,
+            width: 100,
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+          }}
+        >
+          <Text
+            css={{
+              fontWeight: 'bold',
+              width: 50,
+              textAlign: 'center',
+            }}
+            noOfLines={1}
+          >
+            {fontPropertyData.name}
+          </Text>
+        </Box>
+        {(() => {
```

Comment:
nit: Personally I find inline `switch` statements in JSX to be pretty hard to read — I feel like:

```
{ fontProperty === 'fontSize && <FontSizeData /> }
{ fontProperty === 'fontWeight' && <FontWeightData /> }
```

gets the job done!